olive/auto_optimizer/regulate_mixins.py

```diff
@@ -103,22 +103,6 @@ def _regulate_fp16(self, pass_config, pass_flows):
 
         return pass_config, pass_flows
 
-    def regulate_data_config(self, pass_config, pass_flows):
```

Comment:
The auto optimization still requires the data config to run pass/evaluations.

We may need to keep this part. At the same time, could you help update the logics of fetching data_configs.

**Previous**: only use the input model's data config
![image](https://github.com/microsoft/Olive/assets/13343117/d6858f34-2483-4e8c-baa7-fda4461002ea)

**Update to**: 1. if in auto opt model, only one data config can be in data_configs. Otherwise, raise the value error. 2. get this one data_config in `regulate_data_config`.

Reply:
Done!